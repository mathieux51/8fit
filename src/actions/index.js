export const setField = field => ({
  type: "SET_FIELD",
  field
})
