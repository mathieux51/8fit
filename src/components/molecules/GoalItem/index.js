import React from "react"
import { TouchableOpacity, Image, View } from "react-native"
import styled from "styled-components"
import PropTypes from "prop-types"
import TextBold from "atoms/TextBold"
import TextRegular from "atoms/TextRegular"
import images from "src/assets/img"

const Container = styled(TouchableOpacity)`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background: ${({ theme }) => theme.white};
  border: 1px solid ${({ theme }) => theme.lGrey};
  border-radius: 5px;
  padding: 20px;
  width: 90%;
  height: 100px;
  shadow-color: ${({ theme }) => theme.black};
  shadow-offset: 0 3px;
  shadow-opacity: 0.15;
  shadow-radius: 2px;
`

const TextContainer = styled(View)``

const Title = styled(TextBold)`
  margin-bottom: 5px;
`

const Description = styled(TextRegular)`
  margin-top: 5px;
`

const StyledImage = styled(Image)`
  width: 7px;
  height: 12px;
`

const GoalItem = props => (
  <Container
    style={props.style}
    onPress={evt => props.onPress(evt, props.title)}
  >
    <TextContainer>
      <Title>{props.title}</Title>
      <Description>{props.description}</Description>
    </TextContainer>
    <StyledImage source={images.chevronRight} />
  </Container>
)

GoalItem.propTypes = {
  style: PropTypes.array,
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
}

export default GoalItem
