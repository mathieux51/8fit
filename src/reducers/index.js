import { combineReducers } from "redux"
import nav from "./nav"
import user from "./user"

const rootReducer = combineReducers({
  nav,
  user
})

export default rootReducer
