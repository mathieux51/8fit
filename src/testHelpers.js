import React from "react"
import { ThemeProvider } from "styled-components"
import theme from "src/styles/theme"

export function wrappeWithThemeProvider(node) {
  return <ThemeProvider theme={theme}>{node}</ThemeProvider>
}
