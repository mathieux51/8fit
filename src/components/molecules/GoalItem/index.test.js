import React from "react"
import GoalItem from "./"
import { shallow } from "enzyme"
import { wrappeWithThemeProvider } from "src/testHelpers"

describe("<GoalItem />", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        wrappeWithThemeProvider(
          <GoalItem
            title="Lose weight"
            description="Burn fat & get lean"
            onPress={() => {}}
          />
        )
      )
    ).toMatchSnapshot()
  })
})
