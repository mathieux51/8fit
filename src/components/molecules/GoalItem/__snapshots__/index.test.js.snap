// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoalItem /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "black": "#292C2F",
        "bold": 20,
        "boldBig": 24,
        "boldBigger": 32,
        "green": "#26D88C",
        "grey": "#959798",
        "lGrey": "#EAEAEA",
        "medium": 12,
        "regular": 16,
        "regularSmall": 12,
        "white": "#ffffff",
      }
    }
  >
    <GoalItem
      description="Burn fat & get lean"
      onPress={[Function]}
      title="Lose weight"
    />
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "description": "Burn fat & get lean",
      "onPress": [Function],
      "title": "Lose weight",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "description": "Burn fat & get lean",
        "onPress": [Function],
        "title": "Lose weight",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
